backend/README.md

```diff
@@ -1,4 +1,60 @@
 # Pizza Backend
+
 ## Design
 
-We are using Go-lang for the backend. More specifically the [Gin](https://github.com/gin-gonic/gin) web framework for the API, and plans for [Gorilla](https://github.com/gorilla/websocket) as the web socket.
\ No newline at end of file
+We are using Go-lang for the backend. More specifically the [Gin](https://github.com/gin-gonic/gin) web framework for the API, and plans for [Gorilla](https://github.com/gorilla/websocket) as the web socket.
+
+## Routes
+
+We will require routes for the user to be able to enroll into classes, make posts, reply to posts, etc. These should all be grouped under a user route, where we use either the users name or id
+in the route itself for the ability to "enroll" them into a class.
+
+### Structure
+
+user_router := router.Group("/user"){
+
+    user_router.POST("/register", handleUserRegister) // The request respondst to the url matching "/user/register" and should include
+    // information such as first_name, last_name, email. How is user_id being generated? will it be given to us in the request or is is
+    // it made automatically in the database and sent to the front end in a different call
+
+    user_router.POST("/login", handleUserLogin) // The request responds to the url matching "/user/login" and should include
+    // parameters we require for authentication (this will be done later), we can send the user_id to the front end as a response here
+
+    Should we add a reset password for a user route when we add authentication ?
+
+}
+
+The following are all routes for a specific user based on their user_id. Certain routes should only be accessible to professors, e.g. course creation, and archive course.
+
+specific_user_router := router.Group("/user/:user_id"){
```

Comment:
can we do uid?
not as readable but it looks nicer 😎 

similarly, cid for course_id & tid for thread_id

Q: @krishchow 
is golang pothole case or camel case?
the naming for the id variables follows pothole case but in other instances of this file we use camelcase.